test/extended/images/helper.go

```diff
@@ -517,20 +519,36 @@ func MirrorBlobInRegistry(oc *exutil.CLI, dgst digest.Digest, repository string,
 	if err != nil {
 		return err
 	}
-	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s/v2/%s/blobs/%s", registryURL, repository, dgst.String()), nil)
-	if err != nil {
-		return err
-	}
 	token, err := oc.Run("whoami").Args("-t").Output()
 	if err != nil {
 		return err
 	}
-	req.Header.Set("range", "bytes=0-1")
-	req.Header.Set("Authorization", "Bearer "+token)
-	c := http.Client{}
-	resp, err := c.Do(req)
-	if err != nil {
-		return err
+
+	var (
+		req    *http.Request
+		resp   *http.Response
+		getErr error
+	)
+	for _, schema := range []string{"https", "http"} {
```

Comment:
Instead of having logic here from the ordering of these in the list, could make it more explicit:

```go
func MirrorBlobInRegistry(oc *exutil.CLI, dgst digest.Digest, repository string, timeout time.Duration) error {
    // ...
	c := http.Client{
		Transport: knet.SetTransportDefaults(&http.Transport{
			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
		}),
	}

	func getBlob(schema string) (*http.Request, *http.Response, error) {
		req, err := http.NewRequest("GET", fmt.Sprintf("%s://%s/v2/%s/blobs/%s", schema, registryURL, repository, dgst.String()), nil)
		if err != nil {
			return req, nil, err
		}
		req.Header.Set("range", "bytes=0-1")
		req.Header.Set("Authorization", "Bearer "+token)
		resp, err := c.Do(req)
		return req, resp, err
	}

	var (
		req    *http.Request
		resp   *http.Response
		getErr error
	)
	if req, resp, getErr = getBlob("https"); getErr != nil {
		// fallback to http (why is this appropriate?)
		if req, resp, getErr = getBlob("http"); getErr != nil {
			fmt.Fprintf(g.GinkgoWriter, "failed to %s %s: %v (%#+v)\n", req.Method, req.URL, err, err)
			return getErr
		}
	}
    // ...
}
```

Reply:
Good suggestion. Rewritten with a slight modification.